library/microbenchmarks/gpu/gpu_burn/__init__.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+__all__ = ['GpuBurnBase']
+
+
+class GpuBurnBase(rfm.RegressionTest, pin_prefix=True):
+    '''Base class for the GPU Burn test.
+
+    -- Compile --
+    The tests sources can be compiled for both CUDA and HIP. This is set with
+    the `gpu_build` variable, which must be set by a derived class. This
+    source code can also be compiled for a specific device architecture by
+    defining the `gpu_arch` variable.
+
+    -- Run --
+    The variables required by the run stage are:
+     - executable_opts: Defines the options to be passed to the executable.
+     - num_tasks: Number of tasks to use for this test.
+
+    -- Sanity --
+    Checks that the output matches the number of num_gpus_per_node*num_tasks.
+
+    -- Performance --
+    The performance patterns are:
+     - perf: Gflops/s achieved on average for the duration of the run.
+     - temp: the GPU's temperature in degC after the run.
+
+    '''
+
+    # GPU build variables
+    gpu_build = variable(str)
+    gpu_arch = variable(str, type(None), value=None)
+
+    # Required variables
+    executable_opts = required
+    num_tasks = required
+
+    descr = 'GPU burn test'
+    build_system = 'Make'
+    executable = './gpu_burn.x'
+    num_tasks_per_node = 1
```

Comment:
This test should set or require the `num_gpus_per_node`. When I run a simple flexible test, I get a sanity error, because `num_tasks_assigned` is 0. If I set `num_gpus_per_node`, then it passes.